src/contract.rs

```diff
@@ -1378,16 +1485,130 @@ mod tests {
             .query_wasm_smart(&dao_addr, &config_query)
             .unwrap();
 
-        let cw20 = Cw20Contract(cw20_addr);
         assert_eq!(
             res,
             ConfigResponse {
                 config: Config {
                     threshold,
                     max_voting_period: voting_period,
-                    cw20_addr: cw20,
+                    cw20_addr: Cw20Contract(cw20_addr.clone()),
+                    proposal_deposit: ProposalDeposit {
+                        amount: Uint128::zero(),
+                        token_address: Cw20Contract(cw20_addr),
+                    }
                 },
             }
         )
     }
+
+    #[test]
+    fn test_proposal_deposit_works() {
```

Comment:
Can be a separate ticket, but we should add a test to check the when the proposal deposit is 0, it all works without having to set an allowance. Will make a ticket.

Reply:
All the existing tests have the proposal deposit set to zero, so we should be good from a coverage perspective. Thought could make sense to have a explicit test to document the requirement.